subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationIdentifierRegistry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.operations;
+
+import org.gradle.api.Nullable;
+
+public class BuildOperationIdentifierRegistry {
```

Comment:
This actually duplicates the work that `DefaultBuildOperationExecutor` does. We should merge them so that `DefaultBuildOperationExecutor` is the single point of reference for what the current operation is.